docs/source/external_exporters.rst

```diff
@@ -0,0 +1,156 @@
+.. _external_exporter
+
+Using external exporters
+========================
+
+.. versionadded:: 4.2
+
+    You can now use the ``--to`` flag to use custom export formats defined in other
+    libraries.
+
+
+The command-line syntax to run the ``nbconvert`` script is::
+
+  $ jupyter nbconvert --to FORMAT notebook.ipynb
+
+This will convert the Jupyter document file ``notebook.ipynb`` into the output
+format designated by the ``FORMAT`` string as explained below.
+
+A few built-in formats are available by default: `html`, `pdf`,
+`script`, `latex`. Each of these has its own _exporter_ with many configuration
+options that can be extended. Having the option to point to a different _exporter_ 
+allows authors to create their own fully customized templates. 
+
+A custom _exporter_ must be a globally importable python object. We recommend that
+these be distributed as python libraries.
+
+To export using an exporter from an external library, use the `fully qualified`
+name of this exporter on the command line as the argument of the ``--to`` flag::
```

Comment:
I would be fine using entry points, though the approach with DottedObjectNames has IMHO a few advantages:
 - you don't require to install a package you just need the thing to be importable, so it can be a local file to CWD.
 - and understand how to use entrypoint, which mean you don't have to understand how to package something.
 - it forces namespaces. Otherwise N people will try to grab the `--to docx` or others.
 - it does not allow dynamic generation of exporters. 

So in general I would lean toward keep this as is (full qualified named as argument) **for now**, and add ability to register entrypoints on top of that.

So basically provide shorthands as entrypoints (docx, rst, md, py, script, html, slides...) and also allow fully qualified names. Which would be easy to distinguish because full qualified names have dots in their names.